layout/layout_test.go

```diff
@@ -1224,3 +1147,431 @@ func testImage(t *testing.T, when spec.G, it spec.S) {
+				})
+			})
+
+			when("#FromBaseImageIndexInstance", func() {
+				it.Before(func() {
+					localIndex := h.ReadImageIndex(t, baseIndexPath)
+
+					idx, err = layout.NewIndex("busybox-multi-platform", tmpDir, imgutil.FromBaseImageIndexInstance(localIndex))
+					h.AssertNil(t, err)
+
+					localPath = filepath.Join(tmpDir, "busybox-multi-platform")
+				})
+
+				it("manifests from base image index instance are saved on disk", func() {
+					err = idx.Save()
+					h.AssertNil(t, err)
+
+					// assert linux/amd64 and linux/arm64 manifests were saved
+					index := h.ReadIndexManifest(t, localPath)
+					h.AssertEq(t, len(index.Manifests), 2)
+					h.AssertEq(t, index.Manifests[0].Digest.String(), "sha256:4be429a5fbb2e71ae7958bfa558bc637cf3a61baf40a708cb8fff532b39e52d0")
+					h.AssertEq(t, index.Manifests[1].Digest.String(), "sha256:8a4415fb43600953cbdac6ec03c2d96d900bb21f8d78964837dad7f73b9afcdc")
+				})
+			})
+		})
+	})
+
+	when("#Add", func() {
+		var (
+			imagePath         string
+			fullBaseImagePath string
+		)
+
+		it.Before(func() {
+			imagePath, err = os.MkdirTemp(tmpDir, "layout-test-image-index")
+			h.AssertNil(t, err)
+
+			fullBaseImagePath = filepath.Join(testDataDir, "busybox")
+		})
+
+		when("index is created from scratch", func() {
+			it.Before(func() {
+				repoName := newRepoName()
+				idx = setUpImageIndex(t, repoName, tmpDir)
+				localPath = filepath.Join(tmpDir, repoName)
+			})
+
+			when("manifest in OCI layout format is added", func() {
+				var editableImage imgutil.EditableImage
+				it.Before(func() {
+					editableImage, err = layout.NewImage(imagePath, layout.FromBaseImagePath(fullBaseImagePath))
+					h.AssertNil(t, err)
+				})
+
+				it("adds the manifest to the index", func() {
+					err = idx.Add("busybox", imgutil.WithLocalImage(editableImage))
+					h.AssertNil(t, err)
+
+					// manifest was added
+					index := h.ReadIndexManifest(t, localPath)
+					h.AssertEq(t, len(index.Manifests), 1)
+				})
+			})
+		})
+
+		when("index exists on disk", func() {
+			when("#FromBaseImageIndex", func() {
+				it.Before(func() {
+					idx = setUpImageIndex(t, "busybox-multi-platform", tmpDir, imgutil.FromBaseImageIndex(baseIndexPath))
+					localPath = filepath.Join(tmpDir, "busybox-multi-platform")
+				})
+
+				when("manifest in OCI layout format is added", func() {
+					var editableImage imgutil.EditableImage
+					it.Before(func() {
+						editableImage, err = layout.NewImage(imagePath, layout.FromBaseImagePath(fullBaseImagePath))
+						h.AssertNil(t, err)
+					})
+
+					it("adds the manifest to the index", func() {
+						err = idx.Add("busybox", imgutil.WithLocalImage(editableImage))
+						h.AssertNil(t, err)
+
+						index := h.ReadIndexManifest(t, localPath)
+
+						// manifest was added
+						// initially it has 2 manifest + 1 new
+						h.AssertEq(t, len(index.Manifests), 3)
+					})
+				})
+			})
+		})
+	})
+
+	when("#Push", func() {
```

Comment:
Should this go in the remote package?

Reply:
I know it is confusing, but we wanted to allow to push an `layout` and `remote` index